Modules/Filtering/DistanceMap/include/itkDirectedHausdorffDistanceImageFilter.h

```diff
@@ -185,7 +185,7 @@ class ITK_TEMPLATE_EXPORT DirectedHausdorffDistanceImageFilter : public ImageToI
   DistanceMapPointer m_DistanceMap;
 
   Array<RealType>       m_MaxDistance;
-  Array<IdentifierType> m_PixelCount;
+  Array<IdentifierType> m_PixelCount = Array<RealType>::Fill(NumericTraits<IdentifierType>::ZeroValue());
```

Comment:
They (_m_PixelCount_, _m_MaxDistance_, _m_Sum_) are initialized to size of number of threads  in _::BeforeThreadedGenerateData()_ .  Here they have zero size. I guess this can not compile too, AFAIK.

Reply:
You are right; I had a look at `BeforeThreadedGenerateData()` but here I assumed that the same thinking could not be applied. At the same time, I questioned myself about the dynamic analysis warning if the array has no allocated size at this point/when calling it for a newly created class instance to print its members.  I will need to investigate more.

Reply:
Sorry for noise.  I would recommend simply remove internal temporaries from print functions in all affected classes to get rid of Valgrind defects, that print function is IMHO not for debug, more to view parameters (defaults or set by a user). Alternatively, may be it could be possible use `if (GetSize() > 0)` (not tested, not sure) in print function for arrays that are _purposely_ not initialized in constructor, if e.g. size is assumed to be number of threads, like above arrays. There are tests using print function directly after New(), there is IMHO no useful information from internal temporaries at that point, even if they are initialized somehow.